src/SSL.cpp

```diff
@@ -37,6 +43,79 @@
 void MumbleSSL::initialize() {
 	// Let Qt initialize OpenSSL...
 	QSslSocket::supportsSsl();
+
+	// Check that we aren't in a situation where
+	// Qt has dynamically loaded *another* copy of
+	// of libssl and/or libcrypto alongside the copy
+	// we link against ourselves.
+	//
+	// This can happen when Qt is built without the
+	// -openssl-linked configure option. When Qt is
+	// built without -openssl-linked, it will try
+	// to dynamically load the OpenSSL libraries.
+	// When Qt dynamically loads OpenSSL and there
+	// are multiple version of libcrypto and libssl
+	// available on the system, it is possible for
+	// Qt to load another version of OpenSSL in
+	// addition to the one our binary is linked
+	// against. Typically, this will happen if
+	// Mumble or Murmur are linked against
+	// an older version, and a newer version is
+	// available on the system as well.
+	/
+	// If we're in a situation where two version of
+	// OpenSSL's libraries are loaded into our process,
+	// we abort immediately. Things WILL go wrong.
+	MumbleSSL::qsslSanityCheck();
+}
+
+// Check that we haven't loaded multiple copies of OpenSSL
+// into our process by accident.
+void MumbleSSL::qsslSanityCheck() {
+#ifdef Q_OS_LINUX
+	struct link_map *lm = NULL;
+	void *self = dlopen(NULL, RTLD_NOW);
+	if (self == NULL) {
+		qFatal("SSL: could not dlopen program binary");
+	}
+
+	if (dlinfo(self, RTLD_DI_LINKMAP, &lm) == -1) {
+		qFatal("SSL: unable to acquire link_map: %s", dlerror());
+	}
+	if (lm == NULL) {
+		qFatal("SSL: link_map is NULL");
+	}
+
+	QStringList libssl;
+	QStringList libcrypto;
+
+	while (lm != NULL) {
+		QString name = QString::fromLocal8Bit(lm->l_name);
+		if (name.contains(QLatin1String("libssl"))) {
+			libssl << name;
+		} else if (name.contains(QLatin1String("libcrypto"))) {
+			libcrypto << name;
+		}
+		lm = lm->l_next;
+	}
+
+	bool ok = true;
+	if (libcrypto.size() == 0 || libssl.size() == 0) {
+		qFatal("SSL library query failed: %i libcrypto's and %i libssl's found.", libcrypto.size(), libssl.size());
+	}
+	if (libssl.size() > 1) {
+		qWarning("Found multiple libssl.so copies in binary: %s", qPrintable(libssl.join(QLatin1String(", "))));
```

Comment:
Since this one and the one below are unrecoverable ```qCritical``` might be a better fit. Otherwise the "Aborting due to previous errors" is a bit strange with only warnings coming before it.